core/tests/test_apps.zcash.unified_addresses.py

```diff
@@ -0,0 +1,91 @@
+from common import *
+
+from apps.common import coininfo
+from apps.zcash import unified_addresses
+from trezor.enums.ZcashReceiverTypecode import P2PKH, P2SH, SAPLING, ORCHARD
+
+# source: https://github.com/jarys/zcash-test-vectors/blob/trezor/unified_address.py
```

Comment:
I found some official-looking test vectors at https://github.com/zcash-hackworks/zcash-test-vectors/blob/master/test-vectors/json/unified_address.json. It looks like the ones in this file are different. Do these also come from some official source?

Reply:
I generated test vector by the same official library (zcash-test-vectors) with two minor changes:

- I generated ALL possible combinations of receivers. (The generated 10 random combinations.)
- I serialized test vectors differently to load them more easily in python.

Is it ok?

Reply:
> - I generated ALL possible combinations of receivers. (They generated 10 random combinations.)

Generating more combinations is certainly good work. It would be nicer though if our test vectors were a superset of the official ones.

> - I serialized test vectors differently to load them more easily in python.

Absolutely OK.

Reply:
updated to official test vectors [163f493](https://github.com/trezor/trezor-firmware/pull/2398/commits/163f493d46ecf31c7f10077a1a9b102145ad6cae)

Reply:
While testing on new testvectors I found a mistake :-( I reversed receivers order (which should be ascending). It is repaired now. I got confused by receivers priority order (which is descending).

Reply:
[42f4215](https://github.com/trezor/trezor-firmware/pull/2398/commits/42f42155148673156a5275cead794d5ee18cba12)